google/resource_compute_forwarding_rule.go

```diff
@@ -665,32 +642,11 @@ func expandComputeForwardingRuleSubnetwork(v interface{}, d TerraformResourceDat
 }
 
 func expandComputeForwardingRuleTarget(v interface{}, d TerraformResourceData, config *Config) (interface{}, error) {
-	// This method returns a full self link from a partial self link.
-	if v == nil || v.(string) == "" {
-		// It does not try to construct anything from empty.
-		return "", nil
-	} else if strings.HasPrefix(v.(string), "https://") {
-		// Anything that starts with a URL scheme is assumed to be a self link worth using.
-		return v, nil
-	} else if strings.HasPrefix(v.(string), "projects/") {
-		// If the self link references a project, we'll just stuck the compute v1 prefix on it.
-		return "https://www.googleapis.com/compute/v1/" + v.(string), nil
-	} else if strings.HasPrefix(v.(string), "regions/") || strings.HasPrefix(v.(string), "zones/") {
-		// For regional or zonal resources which include their region or zone, just put the project in front.
-		url, err := replaceVars(d, config, "https://www.googleapis.com/compute/v1/projects/{{project}}/")
-		if err != nil {
-			return nil, err
-		}
-		return url + v.(string), nil
-	}
-	// Anything else is assumed to be a regional resource, with a partial link that begins with the resource name.
-	// This isn't very likely - it's a last-ditch effort to extract something useful here.  We can do a better job
-	// as soon as MultiResourceRefs are working since we'll know the types that this field is supposed to point to.
-	url, err := replaceVars(d, config, "https://www.googleapis.com/compute/v1/projects/{{project}}/regions/{{region}}/")
+	f, err := parseRegionalFieldValue("targetPools", v.(string), "project", "region", "zone", d, config, true)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Invalid value for target: %s", err)
 	}
-	return url + v.(string), nil
+	return f.RelativeLink(), nil
```

Comment:
Hmm I wonder why this was like this in the first place. Do you think it was an oversight or do you think it needs to be that way for some reason (I remember seeing some resources that needed a full URL to be passed and not just a path)

Reply:
moved this to https://github.com/GoogleCloudPlatform/magic-modules/pull/1683, will resolve over there